pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/StoolUploadRoute.java

```diff
@@ -0,0 +1,193 @@
+import org.broadinstitute.dsm.exception.FileColumnMissing;
+import org.broadinstitute.dsm.exception.FileWrongSeparator;
+import org.broadinstitute.dsm.exception.UploadLineException;
+import org.broadinstitute.dsm.model.StoolUploadObject;
+import org.broadinstitute.dsm.security.RequestHandler;
+import org.broadinstitute.dsm.statics.DBConstants;
+import org.broadinstitute.dsm.statics.RoutePath;
+import org.broadinstitute.dsm.statics.UserErrorMessages;
+import org.broadinstitute.dsm.util.SystemUtil;
+import org.broadinstitute.dsm.util.UserUtil;
+import org.broadinstitute.lddp.handlers.util.Result;
+import spark.QueryParamsMap;
+import spark.Request;
+import spark.Response;
+
+@Log4j2
+public class StoolUploadRoute extends RequestHandler {
+
+    private static final String PARTICIPANT_ID = "participantId";
+    private static final String MF_BARCODE = "mfBarcode";
+    private static final String RECEIVE_DATE = "receiveDate";
+
+    @Override
+    protected Object processRequest(Request request, Response response, String userId) throws Exception {
+        QueryParamsMap queryParams = request.queryMap();
+        String realm;
+
+        if (queryParams.value(RoutePath.REALM) != null) {
+            realm = queryParams.get(RoutePath.REALM).value();
+        } else {
+            throw new RuntimeException("No realm query param was sent");
+        }
+
+        String userIdRequest = UserUtil.getUserId(request);
+
+        if (UserUtil.checkUserAccess(realm, userId, "kit_upload", userIdRequest)) {
+            String kitTypeName;
+
+            AtomicReference<String> kitUploadReason = new AtomicReference<>();
+            AtomicReference<String> shippingCarrier = new AtomicReference<>();
+            if (queryParams.value(RoutePath.KIT_TYPE) != null) {
+                kitTypeName = queryParams.get(RoutePath.KIT_TYPE).value();
+            } else {
+                throw new RuntimeException("No kitType query param was sent");
+            }
+            if (queryParams.value("reason") != null) {
+                kitUploadReason.set(queryParams.get("reason").value());
+            }
+            if (queryParams.value("carrier") != null) {
+                shippingCarrier.set(queryParams.get("carrier").value());
+            }
+
+
+            HttpServletRequest rawRequest = request.raw();
+            String content = SystemUtil.getBody(rawRequest);
+
+            AtomicBoolean uploadAnyway = new AtomicBoolean(false);
+            if (queryParams.value("uploadAnyway") != null) {
+                uploadAnyway.set(queryParams.get("uploadAnyway").booleanValue());
+            }
+
+            try {
+                List<StoolUploadObject> stoolUploadContent = null;
+                if (uploadAnyway.get()) {
+                    stoolUploadContent = Arrays.asList(new Gson().fromJson(content, StoolUploadObject[].class));
+                } else {
+                    try {
+                        stoolUploadContent = isFileValid(content, realm);
```

Comment:
as far as I know `isFileValid` already exists in `KitUploadRoute` instead of duplicating, you could introduce the class which will wrap all file validation logic and use object of that class in both of the case.